PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/resources/META-INF/ruleConfigs.json

```diff
@@ -64,5 +67,16 @@
       "methods_to_check": "getCompletions",
       "anti_pattern_message": "getCompletions API detected. Use the getChatCompletions API instead."
     }
+  },
+  "UpdateCheckpointAsyncCheck": {
+    "methods_to_check": [
+      "updateCheckpointAsync",
+      "block",
+      "subscribe"
+    ],
+    "anti_pattern_message_map": {
+      "no_block": "Calling updateCheckpointAsync() without block() will not do anything, use `block()` or `block` operator with a timeout, or consider using the synchronous version `updateCheckpoint().",
```

Comment:
I think this checker either needs to be 2 checkers or needs to be architected differently.

You could have one visitor that checks for `.block()`, another that checks for `.subscribe()`.  The keys in this map aren't useful because you hard code their look up anyways.

Reply:
the current implementation looks for UpdateCheckpointsAsync then looks into the methods being invoked with it -- I store the keys in an anti-pattern message map to access in the rule

The only one I don't need to hardcode the lookup is with subscribe since once the visitor finds the "subscribe" method, the methodName can be used as a key

Just that it wouldn't work that way as the absence of block/block(timeout) doesn't give me a key to work with